### PR TITLE
Add ruby jwt decoding/validation example

### PR DESCRIPTION
**Pull request by cgomes-handy**

### CODE REVIEW

**Comment by joealba** [comment]
@cgomes-handy Could you write this as a plain Ruby object rather than a Rails API controller?  I think that would make it easier to follow

**Comment by cgomes-handy** [comment]
@joealba Updated.

File: examples/decode.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        iss: "expected_issuer", 
+        verify_iss: true)
+      rescue JWT::DecodeError => e
+        log(e)
+      end
+    @decoded_jwt.present?
+  end
+
+  def jwt
+    request.authorization.to_s[/(?<=\ABearer) +.*\z/].try!(:strip)
```
Comment (joealba) [inline examples/decode.rb:25]: Here's another Rails abstraction we can make explicit in this example code:  https://github.com/rails/rails/blob/1a240cf83e03527567dea68f3c6bea5301d90113/actionpack/lib/action_dispatch/http/request.rb#L404

File: examples/decode.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module Authorization
+  #validation using JWT gem
```
Comment (joealba) [inline examples/decode.rb:2]: `require 'jwt' # https://github.com/jwt/ruby-jwt#readme` at the top to make it easier for someone to trace the gem

File: examples/decode.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  def validate_jwt
+    return if jwt_valid?
+    render json: error
+  end
+
+  def jwt_valid?
+    @decoded_jwt ||=
```
Comment (joealba) [inline examples/decode.rb:9]: Make this a non-memoized local variable?  Or eliminate it?

File: examples/decode.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      begin
+        JWT.decode(
+        jwt, 
+        public_key, 
+        true, 
+        algorithm: "RS256", 
+        iss: "expected_issuer", 
+        verify_iss: true)
+      rescue JWT::DecodeError => e
+        log(e)
```
Comment (joealba) [inline examples/decode.rb:19]: Replace this with a comment instead

**Comment by cgomes-handy** [comment]
@joealba updated

File: examples/decode.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        verify_iss: true)
+      rescue JWT::DecodeError => e
+        # handle error
+      end
+    decoded_jwt.present?
+  end
+
+  def jwt
+    pattern = /^Bearer /
+    header = request.headers["Authorization"]
```
Comment (joealba) [inline examples/decode.rb:22]:
```suggestion
    header = request.headers["HTTP_AUTHORIZATION"] # Header name may be different based on your environment -- ex: X_HTTP_AUTHORIZATION
```

Comment (cgomes-handy) [reply examples/decode.rb]: updated

**Review by joealba — APPROVED** [review]

**Pull request merged**